db_test.go

```diff
@@ -1145,6 +1146,42 @@ func Test_getRecordFromKey(t *testing.T) {
 	}
 }
 
+func TestDB_ReadErrThenWriteDeadLock(t *testing.T) {
```

Comment:
That case seems not enough. In that case, seems all tx goes well,  and we looking is there any deadlock.
And maybe we can add one more test case. Which is, one tx return error, is that impact the other tx? 

That's my opinion.

Reply:
I modified the test. I let the error happen and then test if the error affects subsequent reads/writes.